File: Backend/DTOs/User/UserProfileGetDTO.cs
﻿namespace Learnz.DTOs;

public class UserProfileGetDTO
{
    public string Username { get; set; }
    public string Firstname { get; set; }
    public string Lastname { get; set; }
    public DateTime Birthdate { get; set; }
    public Grade Grade { get; set; }
    public FileFrontendDTO ProfileImage { get; set; }
    public string Information { get; set; }
    public Language Language { get; set; }
    public Subject GoodSubject1 { get; set; }
    public Subject GoodSubject2 { get; set; }
    public Subject GoodSubject3 { get; set; }
    public Subject BadSubject1 { get; set; }
    public Subject BadSubject2 { get; set; }
    public Subject BadSubject3 { get; set; }
    public bool DarkTheme { get; set; }
}

File: Frontend/src/app/DTOs/User/UserProfileGetDTO.ts
import { LearnzFileFrontendDTO } from "../File/LearnzFileFrontendDTO";

export interface UserProfileGetDTO {
  username: string;
  firstname: string;
  lastname: string;
  birthdate: Date;
  grade: number;
  profileImage: LearnzFileFrontendDTO;
  information: string;
  language: number;
  goodSubject1: number;
  goodSubject2: number;
  goodSubject3: number;
  badSubject1: number;
  badSubject2: number;
  badSubject3: number;
  darkTheme: boolean;
}